src/main/java/org/apache/maven/plugins/clean/Cleaner.java

```diff
@@ -115,9 +135,102 @@ public void delete( File basedir, Selector selector, boolean followSymlinks, boo
 
         File file = followSymlinks ? basedir : basedir.getCanonicalFile();
 
+        if ( selector == null && !followSymlinks && fastDir != null )
+        {
+            // If anything wrong happens, we'll just use the usual deletion mechanism
+            if ( fastDelete( file ) )
+            {
+                return;
+            }
+        }
+
         delete( file, "", selector, followSymlinks, failOnError, retryOnError );
     }
 
+    private boolean fastDelete( File baseDir )
+    {
+        // Handle the case where we use ${maven.multiModuleProjectDirectory}/target/.clean for example
+        if ( fastDir.getAbsolutePath().startsWith( baseDir.getAbsolutePath() ) )
```

Comment:
Can't we use `Path` for this? String comparison and Windows...

Reply:
I don't think using `Path` will change anything...
That would look like:
```
        fastDir.toPath().startsWith( baseDir.toPath() )
```
but I think the semantics will be exactly the same.

Reply:
It will, since it will compare according to the semantics of the underyling FS/OS. Case, separators, etc.